src/app/core/common-components/json-editor/json-editor.component.ts

```diff
@@ -29,12 +32,11 @@ import { MatFormFieldControl } from "@angular/material/form-field";
   ],
 })
 export class JsonEditorComponent extends CustomFormControlDirective<object> {
-  @Output() isValidChange = new EventEmitter<boolean>();
-
   @ViewChild("json", { static: true }) json!: ElementRef<HTMLDivElement>;
 
+  @Input() formControl!: FormControl;
```

Comment:
The `CustomFormControlDirective` already has some logic to integrate with a formControl being passed. I think this should be available through `this.ngControl.control`.

To not interfere with other logic that set errors, I would also recommend that we try to integrate the json-valid check with CustomFormControlDirective's flow, see
 https://github.com/Aam-Digital/ndb-core/blob/0a23155a456ad87f128f7d2924c5432e41a661aa/src/app/core/common-components/basic-autocomplete/custom-form-control.directive.ts#L166

i.e. could you try in the constructor here to create a custom errorStateMatcher that makes an "or" combination of the this.errorStateMatcher.isErrorState() and your json-invalid check?

Reply:
Please check the new implementation and try to break it, @Ayush8923  :sweat_smile:     I think this could work